icinga2/synology-services.conf

```diff
@@ -0,0 +1,104 @@
+    display_name += " at " + host.vars.synology_location
+  }
+
+  // Enable this to make the command run on the monitored host
+  // when different from the Icinga instance.
+  // command_endpoint   = host_name
+
+}
+
+apply Service "Synology system load" {
+  import "generic-service"
+  import "synology-base"
+
+  // Warn if system load is higher than X.
+  vars.synology_mode = "load"
+  vars.synology_warning = host.vars.synology_load1_warning || 5
+  vars.synology_critical = host.vars.synology_load1_critical || 10
+
+  assign where host.vars.check_synology == true || host.vars.os == "DSM"
+}
+
+apply Service "Synology system memory" {
+  import "generic-service"
+  import "synology-base"
+
+  // Warn if available system memory is lower than %.
+  vars.synology_mode = "memory"
+  vars.synology_warning = host.vars.synology_memory_warning || 20
+  vars.synology_critical = host.vars.synology_memory_critical || 10
+
+  assign where host.vars.check_synology == true || host.vars.os == "DSM"
+}
+
+apply Service "Synology disk temperature" {
+  import "generic-service"
+  import "synology-base"
+
+  // Warn if disk drive temperature is higher than °C.
+  // https://www.hdsentinel.com/forum/viewtopic.php?f=32&t=1519
+  vars.synology_mode = "disk"
+  vars.synology_warning = host.vars.synology_hddtemp_warning || 43
+  vars.synology_critical = host.vars.synology_hddtemp_critical || 50
+
+  assign where host.vars.check_synology == true || host.vars.os == "DSM"
+}
+
+apply Service "Synology disk space" {
+  import "generic-service"
+  import "synology-base"
+
+  // Warn if used disk space is higher than %.
+  vars.synology_mode = "storage"
+  vars.synology_warning = host.vars.synology_storage_warning || 85
+  vars.synology_critical = host.vars.synology_storage_critical || 95
+
+  // Needed to be adjusted to work against "Timeout exceeded" errors.
+  check_timeout = 120
+
+  // Check each 15 minutes.
+  check_interval = 900
+
+  assign where host.vars.check_synology == true || host.vars.os == "DSM"
+}
+
+apply Service "Synology update status" {
+  import "generic-service"
+  import "synology-base"
+
+  // Enable checks with assigning a value of "1".
+  // When enabled, report WARNING if update status is "Available".
+  // When enabled, report CRITICAL if update status is "Disconnected" or "Others".
+  vars.synology_mode = "update"
+  vars.synology_warning = host.vars.synology_update_warning || 1
+  vars.synology_critical = host.vars.synology_update_critical || 1
+
+  assign where host.vars.check_synology == true || host.vars.os == "DSM"
+}
+
+apply Service "Synology system temperature" {
+  import "generic-service"
+  import "synology-base"
+
+  // Warn if system temperature is higher than °C.
+  // https://www.tomshardware.com/how-to/how-to-monitor-cpu-temp-temperature
+  vars.synology_mode = "status"
+  vars.synology_warning = host.vars.synology_temperature_warning || 65
+  vars.synology_critical = host.vars.synology_temperature_critical || 80
```

Comment:
I am not sure about those default thresholds. Do you know if this is a temperature sensor attached to the motherboard, or is it actually CPU core temperature?

To provide some empirical evidence to this question, from the perspective of measured values, we are sharing the values from two monitored appliances. One yields a measurement value of 40°C, while the other one has 55°C already.

![image](https://user-images.githubusercontent.com/453543/158014709-e9efb409-1f0c-44b8-b367-93adcb8f3a3c.png)

I didn't exactly follow the advices from https://www.tomshardware.com/how-to/how-to-monitor-cpu-temp-temperature, which would have been 80/95°C. Maybe you have any opinions about this, @wernerfred or @tonkenfo?

Reply:
I just checked another plugin, [Synology status], written in Bash. It employs those default threshold values, which are applied to _both_ system and disk temperatures.
```bash
warningTemperature="50"
criticalTemperature="60"
```

Those values resonate more with my gut feeling. Considering the screenshot above, I believe the system with 55°C might prove to have a failed fan already.

[Synology status]: https://exchange.nagios.org/directory/Plugins/Network-and-Systems-Management/Others/Synology-status/details

Reply:
With the most recent update, the WARNING/CRITICAL threshold values are 50/60°C, as outlined above. I believe using a "better safe than sorry" approach for choosing default values is the right way to go, i.e. raising warnings earlier than later in the default configuration. Individual adjustments can be made anytime, so there is nothing to worry about.

This reasoning now has also been reflected within the inline documentation.
```c++
/**
 * 
 * Finding the right thresholds that fit everyone is not possible. So, in general,
 * the default values are more on the "better safe than sorry" side of the spectrum and
 * should be reconfigured based on individual needs.
 *
 * This is specifically applicable to system and disk temperature metrics, as those are
 * highly dependent on ambient temperature levels and vary hugely between regions on earth
 * and/or temperature conditioning scenarios.
 * 
 **/
```

Let me know if you take a different stance on this.